docs/rackhd/configuration.rst

```diff
@@ -215,3 +220,144 @@ choice. Verify the certificate by restarting on-http and visiting
 **Note:** For information about OpenSSL, see the `OpenSSL documentation`_.
 
 .. _OpenSSL documentation: https://www.openssl.org/docs/
+
+
+Setting up HTTP/HTTPS endpoint
+------------------------------
+
+This section describes how to setup HTTP/HTTPS endpoints in RackHD.
+A endpoint is an instance of HTTP or HTTPS server that serves a group of APIs. User can
+choose to enable authentication or enable HTTPs for each endpoint.
+
+There are currently two group of APIs defined in RackHD:
+
+- the northbound-api-router API group. This is the group of API that is used by users
+- the southbound-api-router API group. This is the group of API that is used by the nodes
+
+.. code-block:: JSON
+
+    [
+        {
+            "address": "0.0.0.0",
+            "port": 8443,
+            "httpsEnabled": true,
+            "httpsCert": "data/dev-cert.pem",
+            "httpsKey": "data/dev-key.pem",
+            "httpsPfx": null,
+            "proxiesEnabled": false,
+            "authEnabled": false,
+            "routers": "northbound-api-router"
+        },
+        {
+            "address": "172.31.128.1",
+            "port": 9080,
+            "httpsEnabled": false,
+            "proxiesEnabled": true,
+            "authEnabled": false,
+            "routers": "southbound-api-router"
+        }
+    ]
+
+.. list-table::
+    :widths: 20 100
+    :header-rows: 1
+
+    * - Parameter
+      - Description
+    * - address
+      - IP/Interface to bind to for HTTP. Typically this is '0.0.0.0'
+    * - port
+      - Local port to use for HTTP. Typically, port 80 for HTTP, 443 for HTTPS
+    * - httpsEnabled
+      - Toggle HTTPS
+    * - httpsCert
+      - Filename of the X.509 certificate to use for TLS. Expected format is PEM.
+        This is optional and only take effect when httpsEnabled flag set to true
+    * - httpsKey
+      - Filename of the RSA private key to use for TLS. Expected format is PEM.
+        This is optional and only take effect when httpsEnabled flag set to true
+    * - httpsPfx
+      - Pfx file containing the SSL cert and private key
+        (only needed if the key and cert are omitted)
+        This is optional and only take effect when httpsEnabled flag set to true
+    * - proxiesEnabled
+      - Toggle Proxies
+    * - authEnabled
+      - Toggle API Authentication
+    * - routers
+      - A single router name or a list of router names.
+        This would only take effect for 1.1 APIs.
+        You can now choose from "northbound-api-router","southbound-api-router" or 
+        ["northbound-api-router", "southbound-api-router"].
+
+Authentication
+-------------------------
+
+This section describes how to enable user authentication in RackHD.
+
+Enable Authentication
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+As mentioned in `Setting up HTTP/HTTPS endpoint`_ section, authentication can be enabled
+or disabled per endpoint basis.
+
+Setting authEnabled flag to true in an endpoint configure will enable authentication for
```

Comment:
Say "Setting the authEnabled flag to true in an endpoint configuration will enable"...